module/advancement/advancement.mjs

```diff
@@ -259,8 +259,8 @@ export default class Advancement {
     const advancement = foundry.utils.deepClone(this.item.system.advancement);
     const idx = advancement.findIndex(a => a._id === this.id);
     if ( idx < 0 ) throw new Error(`Advancement of ID ${this.id} could not be found to update`);
-    foundry.utils.mergeObject(this.data, updates);
-    foundry.utils.mergeObject(advancement[idx], updates);
+    foundry.utils.mergeObject(this.data, updates, { performDeletions: true });
+    foundry.utils.mergeObject(advancement[idx], updates, { performDeletions: true });
     this.item.updateSource({"system.advancement": advancement});
```

Comment:
It *should* now in V10 be possible to do this without merging first:
```js
this.item.updateSource({[`system.advancement.${idx}`]: updates});
```

Reply:
That seems to be saving it as an array, not an object. Does this only work with `DataModel` enforced types?

Reply:
> That seems to be saving it as an object, not an array. Does this only work with `DataModel` enforced types?

Ah, yes. My mistake. It only works when the data model specifically defines an `ArrayField` type. Sorry, maybe one for the future!